torchtext/datasets/imdb.py

```diff
@@ -24,15 +24,9 @@
 @_create_dataset_directory(dataset_name=DATASET_NAME)
 @_wrap_split_argument(('train', 'test'))
 def IMDB(root, split):
-    def generate_imdb_data(key, extracted_files):
-        for fname in extracted_files:
-            *_, split, label, file = Path(fname).parts
-
-            if key == split and (label in ['pos', 'neg']):
-                with io.open(fname, encoding="utf8") as f:
-                    yield label, f.read()
     dataset_tar = download_from_url(URL, root=root,
                                     hash_value=MD5, hash_type='md5')
-    extracted_files = extract_archive(dataset_tar)
-    iterator = generate_imdb_data(split, extracted_files)
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split], iterator)
+
+    extracted_files = ReadFilesFromTar([(os.path.dirname(dataset_tar), open(dataset_tar, 'rb'))])
```

Comment:
@VitalyFedyunin I would like to stack it with HttpReader instead of first downloading it using download_from_url. Could you please suggest how to do this? It seems we do not have functional for ReadFilesFromTar?